app/src/main/java/org/wikipedia/readinglist/AddToReadingListDialog.java

```diff
@@ -201,14 +205,33 @@ private void showCreateListDialog() {
     }
 
     private void addAndDismiss(final ReadingList readingList, final PageTitle title) {
-
         if (readingList.pages().size() >= SiteInfoClient.getMaxPagesPerReadingList()) {
             String message = getString(R.string.reading_list_article_limit_message, readingList.title(), SiteInfoClient.getMaxPagesPerReadingList());
             FeedbackUtil.makeSnackbar(getActivity(), message, FeedbackUtil.LENGTH_DEFAULT).show();
             dismiss();
             return;
         }
 
+        execute(readingList, title);
+    }
+
+    private void addAndDismiss(final ReadingList readingList, final List<PageTitle> titles) {
+        if ((readingList.pages().size() + titles.size()) > SiteInfoClient.getMaxPagesPerReadingList()) {
+            String message = getString(R.string.reading_list_article_limit_message, readingList.title(), SiteInfoClient.getMaxPagesPerReadingList());
+            FeedbackUtil.makeSnackbar(getActivity(), message, FeedbackUtil.LENGTH_DEFAULT).show();
+            dismiss();
+            return;
+        }
+
+        if (titles.size() == 1) {
+            addAndDismiss(readingList, titles.get(0));
+            return;
+        }
+
+        execute(readingList, titles);
+    }
+
+    void execute(final ReadingList readingList, final PageTitle title) {
```

Comment:
Naming: what does this execute?

Reply:
I changed it to `run` and not sure if it is a proper name. Do you have any suggestions for this method?

Reply:
Ideally the name of a function should convey a sense of _what it does_.  Names like `run()`, `doIt()`, `go()`, etc. are pretty much meaningless.  I can see how it's hard to give this function a name because it can do different things when it's overridden. See my follow-up for the best I could come up with.